File: C++/1.자료구조/queue&stack/큐2 18258.cpp
// #include <vector>
// #include <algorithm>
// #include <iostream>
// #include <queue>
// using namespace std;

// int main()
// {
//     ios_base::sync_with_stdio(false);
//     cin.tie(NULL);
//     queue<int> q;
//     int t;
//     cin >> t;
//     int end = -1;
//     while(t--)
//     {
//         string a;
//         cin >> a;
//         if(a=="push")
//         {
//             int x;
//             cin >> x;
//             q.push(x);
//             end = x;
//         }
//         else if(a=="pop")
//         {
//             if(!q.empty())
//             {
//                 int x = q.front();
//                 q.pop();
//                 cout << x << '\n';
//             }
//             else cout << -1 << '\n';
//             if(q.empty()) end = -1;
//         }
//         else if(a=="size")
//         {
//             cout << q.size() << '\n';
//         }
//         else if(a=="empty")
//         {
//             cout << q.empty() << '\n';
//         }
//         else if(a=="front")
//         {
//             if(!q.empty()) cout << q.front() << '\n';
//             else cout << -1 << '\n';
//         }
//         else if(a=="back")
//         {
//             cout << end << '\n';
//         }
//     }
//     return 0;
// }
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    cin >> t;
    deque<int> dq;
    while(t--) {
      string a;
      cin >> a;
      if(a=="push") {
        int b;
        cin >> b;
        dq.push_back(b);
      }
      else if(a=="pop") {
        if(dq.empty()) cout << -1 << '\n';
        else {
          cout << dq.front() << '\n';
          dq.pop_front();
        }
      }
      else if(a=="size") {
        cout << dq.size() << '\n';
      }
      else if(a=="empty") {
        if(dq.empty()) cout << 1 << '\n';
        else cout << 0 << '\n';
      }
      else if(a=="front") {
        if(dq.empty()) cout << -1 << '\n';
        else cout << dq.front() << '\n';
      }
      else if(a=="back") {
        if(dq.empty()) cout << -1 << '\n';
        else cout << dq.back() << '\n';
      }
    }
    return 0;
}

File: Java/1.자료구조/queue&stack/큐2 18258.java
// import java.io.BufferedReader;
// import java.io.IOException;
// import java.io.InputStreamReader;
// import java.util.Deque;
// import java.util.LinkedList;
// import java.util.Queue;
// import java.util.StringTokenizer;

// public class Main {

// 	public static void main(String[] args) throws NumberFormatException, IOException {
// 		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
// 		StringBuilder sb = new StringBuilder();
// 		StringTokenizer st;
// 		Deque<Integer> q = new LinkedList<>();

// 		int n = Integer.parseInt(br.readLine());
// 		for(int i=0;i<n;i++) {
// 			st = new StringTokenizer(br.readLine()," ");
// 			String x = st.nextToken();
// 			if(x.equals("push")) {
// 				int y = Integer.parseInt(st.nextToken());
// 				q.offer(y);
// 			} else if(x.equals("pop")) {
// 				if(q.isEmpty()) sb.append(-1).append("\n");
// 				else {
// 					sb.append(q.peekFirst()).append("\n");
// 					q.pollFirst();
// 				}
// 			} else if(x.equals("size")) {
// 				sb.append(q.size()).append("\n");
// 			} else if(x.equals("empty")) {
// 				if(q.isEmpty()) sb.append(1).append("\n");
// 				else sb.append(0).append("\n");
// 			} else if(x.equals("front")) {
// 				if(q.isEmpty()) sb.append(-1).append("\n");
// 				else sb.append(q.peekFirst()).append("\n");
// 			} else if(x.equals("back")) {
// 				if(q.isEmpty()) sb.append(-1).append("\n");
// 				else sb.append(q.peekLast()).append("\n");
// 			}
// 		}
// 		System.out.println(sb);
// 	}
// }
import java.io.*;
import java.util.*;

public class Main {

    static Deque<Integer> q = new LinkedList<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        for(int i=0;i<n;i++) {
            st = new StringTokenizer(br.readLine()," ");
            String x = st.nextToken();
            if("push".equals(x)) {
                String y = st.nextToken();
                q.offer(Integer.parseInt(y));
            } else if("pop".equals(x)) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(Integer.toString(q.pollFirst())).append("\n");
            } else if("size".equals(x)) {
                sb.append(q.size()).append("\n");
            } else if("empty".equals(x)) {
                sb.append(q.isEmpty()==true?1:0).append("\n");
            } else if("front".equals(x)) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(Integer.toString(q.peekFirst())).append("\n");
            } else if("back".equals(x)) {
                if(q.isEmpty()) sb.append("-1").append("\n");
                else sb.append(Integer.toString(q.peekLast())).append("\n");
            }
        }
        System.out.println(sb.toString());
    }
}

